skrf/circuit.py

```diff
@@ -928,20 +937,110 @@ def X(self) -> np.ndarray:
         There is a numerical bottleneck in this function,
         when creating the block diagonal matrice [X] from the [X]_k matrices.
         """
-        Xks = [self._Xk(cnx) for cnx in self.connections]
+        Xks = [self._Xk(cnx, order) for cnx in self.connections]
 
-        Xf = np.zeros((len(self.frequency), self.dim, self.dim), dtype='complex')
+        Xf = np.zeros((len(self.frequency), self.dim, self.dim), dtype='complex', order=order)
         off = np.array([0, 0])
         for Xk in Xks:
             Xf[:, off[0]:off[0] + Xk.shape[1], off[1]:off[1]+Xk.shape[2]] = Xk
             off += Xk.shape[1:]
 
         return Xf
 
+    @cached_property
+    def X(self) -> np.ndarray:
+        """
+        Return the concatenated intersection matrix [X] of the circuit in C-order.
+
+        It is composed of the individual intersection matrices [X]_k assembled
+        by block diagonal.
+
+        Returns
+        -------
+        X : :class:`numpy.ndarray`
+
+        Note
+        ----
+        There is a numerical bottleneck in this function,
+        when creating the block diagonal matrice [X] from the [X]_k matrices.
```

Comment:
same here with the comment: still valid?

Reply:
The `X` property returns the `[X]` matrix in 'C' order, and thus, the comments regarding the performance bottleneck in block diagonal matrix creation remain pertinent and accurate.